### ISSUE TITLE
Postgres: Found unparsable section: INSERT ... SELECT ... RETURNING in version 3.0.6

### ISSUE THREAD

**Issue by Demianeen** [sqlfluff/sqlfluff#5865]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

[related](https://github.com/sqlfluff/sqlfluff/issues/4626)

Sqlfluff can't parse:

```sql
INSERT INTO users (id, created_at, updated_at, name)
VALUES (1, 2, 3, 4)
RETURNING *;
```

And throws:
`Line 4, Position 1: Found unparsable section: 'RETURNING *;' sqlfluff (PRS) [4, 1]`

### Expected Behaviour

Correct parsing

### Observed Behaviour

PostresQL accepts RETURNING keyword, so the query is valid.

### How to reproduce

1. Add .sqlfluff file:
```toml
[sqlfluff]
dialect = postgres

[sqlfluff:indentation]
indent_unit = tab

[sqlfluff:rules:references.keywords]
ignore_words = name

[sqlfluff.rules.capitalisation.keywords]
capitalisation_policy = "upper"
```
2. Create .sql file with the following code:
```
-- name: CreateUser :one
INSERT INTO users (id, created_at, updated_at, name)
VALUES (1, 2, 3, 4)
RETURNING *;
```

### Dialect

postgres

### Version

3.0.6

### Configuration

```toml
[sqlfluff]
dialect = postgres

[sqlfluff:indentation]
indent_unit = tab

[sqlfluff:rules:references.keywords]
ignore_words = name

[sqlfluff.rules.capitalisation.keywords]
capitalisation_policy = "upper"
```

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by Demianeen** [comment]
I would be happy to submit a PR, but I would need some guidance where to start

**Comment by hiporox** [comment]
I am running into this same issue on version 3.3.1
